blackboxopt/optimizers/botorch_base.py

```diff
@@ -86,12 +118,20 @@ def _acquisition_function_optimizer_factory(
             **kwargs,
         )
 
-    choices = torch.Tensor(
-        [
-            search_space.to_numerical(search_space.sample())
-            for _ in range(kwargs.pop("num_random_choices", 5_000))
-        ]
-    ).to(dtype=torch_dtype)
+    if "num_random_choices" not in kwargs and not space_has_continuous_parameters:
+        # Optimize over the entire discrete search space, if the number of random
+        # choices is not specified
+        choices = torch.from_numpy(
+            _get_numerical_points_from_discrete_space(search_space)
+        ).to(torch_dtype)
+    else:
+        # Optimize over the desired number of samples from the discrete search space
+        choices = torch.Tensor(
+            [
+                search_space.to_numerical(search_space.sample())
+                for _ in range(kwargs.pop("num_random_choices", 5_000))
```

Comment:
Do I see correctly that the default of `5_000` will never be used in the current implementation and could thus be removed? Because if we enter this decision branch, either `num_random_choices in kwargs` or if not, `space_has_continuous_parameters` would need to be true, which would have triggered the return in 111 already.